### DISCUSSION TITLE
`gr` (go to reference) is very slow for `python` code

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#4241] [Q&A]
As shown in the video below, it took several tries to get `gr` (go to reference) to work, i.e. showing the reference of the variable under the cursor in telescope, and this was in a fresh installation of `lazyvim`. 

May I ask if this is normal? If not, what should be done to improve the response of `lsp` related functions for `python` code? 

https://github.com/user-attachments/assets/f9eb085d-44ef-4fe6-9fe5-cb594e036cc1

**Comment by dpetka2001** [comment]
This is entirely dependent on the LSP server itself and the size of the project it has to search through to find references. On a simple program that I tried it was instant. Screecast following

[Screencast 2024-08-02 11:39:03.webm](https://github.com/user-attachments/assets/cdb7f16f-1fe3-4147-9b7a-bfe0820567d8)
